request-body-canonicalization/latest/index.md

```diff
@@ -0,0 +1,279 @@
+
+The key words MAY and MUST in this document are to be interpreted as described in BCP 14 [RFC2119][1] [RFC8174][2] when, and only when, they appear in all capitals, as shown here.
+
+## Terminology
+
+- CDX
+- CDXJ
+- WACZ
+- WARC
+
+## Introduction
+
+### Web Archive Formats (WARC and WACZ)
+
+Web archiving data is often stored in specialized formats, which include a full record of the HTTP network traffic as well as additional metadata. The archived data is often accessed via random-access, loading the appropriate chunks of data based on URLs requested by end users.
+
+Web archiving data is often stored in two key file formats:
+
+1. WARC — A widely accepted [ISO standard][3] used by many institutions around the world for storing web archive data.
+2. WACZ — A new format [developed by Webrecorder][4] for packaging WARCs with other web archive data which supports random-access reads.
+
+Both formats are 'composite' formats, containing smaller amounts of data interspersed with metadata. In the case of WARC, the format consists of concatenated records which are appended one after the other, eg. `cat A.warc B.warc > C.warc`. The WARCs may or may not be gzipped, in which case the result is a multi-member gzip.
+
+WACZ files use the ZIP format, which contains a specialized file and directory layout. ZIP is also a composite format, containing the raw (sometimes compressed) data as well as header data which contains the location files and directories within the ZIP file.
+
+## Web Archive Index Formats (CDX and CDXJ)
+
+Web archive search and retrieval is frequently intermediated by index files of WARC data in the CDX or CDXJ formats. WACZ files contain CDXJ indices, which may or may not be gzipped, within the ZIP file that comprises the WACZ.
+
+### CDX
+
+CDX is a web archive index format developed as part of the Internet Archive's Wayback Machine, where CDX may have been an acronym for Crawl (or Capture) inDeX. A CDX file consists of plain text, with the first line being a legend and each line afterwards describing a web document. More information about how the format works can be found in the [CDX specification][5].
+
+CDX was the precursor to the CDXJ index format.
+
+### Crawl Index JSON (CDXJ)
+
+Crawl Index JSON or [CDXJ](4) provides a standardized way of representing an index to one or more WARC files. It allows applications to quickly locate a given page in a set of archived web content, as well as metadata associated with that page. Each CDXJ entry can be looked up by URL, and contains a JSON payload that can be used for representing information about that URL. It is used in the [WACZ specification][4].
+
+A CDXJ file is a sorted, line oriented plain-text file (optionally GZIP compressed) where each line represents information about a single capture in a web archive collection.
```

Comment:
Single captured what?  Resource?  Whole page and all of its component resources?

Reply:
```suggestion
A CDXJ file is a sorted, line oriented plain-text file (optionally GZIP compressed) where each line represents information about a single captured URL in a web archive collection.
```

Reply:
Alternative could use resource if you prefer? But I think URL might be clearer to a lay user.